scripts/src/models/orders.ts

```diff
@@ -325,9 +325,13 @@ function extendedOrder(origin: OrderOrigin): (typeof Order) & OrderFactory {
 	return Object.assign({}, Order, {
 		"new"(data?: DeepPartial<Order>, ...context: Array<DeepPartial<OrderContext>>): Order {
 			data = Object.assign(
-				{ nonce: Order.DEFAULT_NONCE },
 				data,
 				{ origin });
+
+			if (isNothing(data.nonce)) {
+				(data as Mutable<Order>).nonce = Order.DEFAULT_NONCE;
```

Comment:
you need to make sure no one will alter Order.DEFAULT_NONCE in the future - might be an overkill but simple comment can help